### ISSUE TITLE
Error: the condition has length > 1

### ISSUE THREAD

**Issue by AvoidAcid** [bertolt01/asw.cluster#1]
I run the example code but met the following error:
```r
> my_ASW <- faultlines(data = mydata,  
+                      group.par = "teamid",  
+                      attr.type = my_types,   
+                      attr.weight = my_weights,   
+                      method = "asw",
+                      cores=1)
Error in if (is.na(attr.type)) { : the condition has length > 1
```

Environment:
- `asw.cluster` version 2.10.1
- R version 4.2.2
- System: macOS Ventura 13.0.1

**Comment by RobbeVerresen** [comment]
I also encountered the same error. The issue is that if `attr.type` is not empty, `is.na()` will return a boolean vector of length equal to `length(attr.type)`, which is not compatible with the `if` statement. To fix the error, change it to `if (all(is.na(attr.type)))`. Please note that there are multiple instances of this bug (e.g., `if (is.na(attr.weight)))`) in the code, so be sure to change all of them to debug the code fully.

**Comment by ben3mey** [comment]
Hey @RobbeVerresen! Thank you for your solution! I encounter the same issue and I am not entirely sure what you mean. Do I have to make the changes in the code from the function or is there a solution without changing the code of the function?
I am asking because I am new to R and do not quite know how to change the `faultlines()` function itself. Thank you!

Additionally, I am not able to call the source code of `faultlines()`. `debug(faultlines)` states that the source code is not available, and `body(faultlines)` only shows the usage of the function as it can be found in the help view with `?faultlines`. 
Any help is really much appreciated! :)

**Comment by AvoidAcid** [comment]
> I also encountered the same error. The issue is that if `attr.type` is not empty, `is.na()` will return a boolean vector of length equal to `length(attr.type)`, which is not compatible with the `if` statement. To fix the error, change it to `if (all(is.na(attr.type)))`. Please note that there are multiple instances of this bug (e.g., `if (is.na(attr.weight)))`) in the code, so be sure to change all of them to debug the code fully.

Exactly. I found another solution: just use R versions 3.x. For me, R 3.6.3 worked.

**Comment by AvoidAcid** [comment]
@ben3mey 
This is how I found the code and might be helpful:

First, type `methods(faultlines)` and get this:
```
> methods(faultlines)
[1] faultlines.default*
see '?methods' for accessing help and source code
```

Then, type `getAnywhere(faultlines.default)` to see the source code of `faultlines`. 

Finally, copy the function body and change line number 48 from `if (is.na(attr.type)) {` to `if (missing(attr.type)) {` or something else from RobbeVerresen's answer.

**Comment by bertolt01** [comment]
Dear all, sorry for the delayed response, we fixed the issue that was introduced by R 4.0.0 and went unnoticed. We just updated the package. Re-installing should fix the issue. Thank you @AvoidAcid for providing a fix.
